{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e8bb269f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "from biomart import BiomartServer\n",
    "#import time\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e6aba0a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# DICTIONARY\n",
    "dictionary = {}\n",
    "\n",
    "def gene_dictionary(index,\n",
    "                    gene_name,\n",
    "                    cluster,\n",
    "                    gene_list_id,\n",
    "                    ensembl_gene_id,\n",
    "                    ensembl_transcript_id,\n",
    "                    refseq_mrna,\n",
    "                    hgnc_symbol,\n",
    "                    alias,\n",
    "                    direction):\n",
    "    \n",
    "    dictionary = {\n",
    "        'Index' : index,\n",
    "        'Gene_name' : gene_name,\n",
    "        'Gene_list_id' : gene_list_id,\n",
    "        'Ensembl_gene_id' : ensembl_gene_id,\n",
    "        'Ensembl_transcript_id' : ensembl_transcript_id,\n",
    "        'RefSeq_mRNA_id' : refseq_mrna,\n",
    "        'HGNC_symbol' : hgnc_symbol,\n",
    "        'UP' : cluster,\n",
    "        'DOWN' : cluster,\n",
    "        'Alias' : alias,\n",
    "        'Info' : direction\n",
    "    }\n",
    "    \n",
    "    return dictionary\n",
    "\n",
    "\n",
    "\n",
    "# REGULATION\n",
    "def regulation(gene_name):\n",
    "    if gene_name in downTab:\n",
    "        return 'Down'\n",
    "    elif gene_name in upTab:\n",
    "        return 'Up'\n",
    "    else:\n",
    "        return 'Unknown'\n",
    "    \n",
    "\n",
    "# RESPONSE  \n",
    "def biomartParameters(mgi_symbol):\n",
    "\n",
    "    attributes = ['ensembl_gene_id',\n",
    "                  'ensembl_transcript_id',\n",
    "                  'refseq_mrna']\n",
    "\n",
    "    filters = {'mgi_symbol':[mgi_symbol]}                     # gene_name = mgi_symbol\n",
    "    response = dataset.search({'attributes':attributes,'filters':filters})\n",
    "    \n",
    "\n",
    "# response_convertion\n",
    "    lines = response.text.split(\"\\n\")\n",
    "    lines = list(filter(lambda l: l != \"\",lines))\n",
    "    values = []\n",
    "    for l in lines:\n",
    "        temp = []\n",
    "        raw_values = l.split(\"\\t\")\n",
    "        values.append(raw_values)\n",
    "        \n",
    "# if values not return ????\n",
    "\n",
    "    return values  \n",
    "\n",
    "def biomartHumanOrthologs(mgi_symbol):\n",
    "    attributes = ['hsapiens_homolog_associated_gene_name']\n",
    "    filters = {'mgi_symbol':[mgi_symbol]}\n",
    "    response = dataset.search({'attributes':attributes,'filters':filters})\n",
    "    \n",
    "# response_convertion\n",
    "    lines = response.text.split(\"\\n\")\n",
    "    lines = list(filter(lambda l: l != \"\",lines))\n",
    "    values = []\n",
    "    for l in lines:\n",
    "        temp = []\n",
    "        raw_values = l.split(\"\\t\")\n",
    "        values.append(raw_values)\n",
    "        \n",
    "# if values not return ????\n",
    "\n",
    "    return values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3bc8a756",
   "metadata": {},
   "outputs": [],
   "source": [
    "# clusters_file\n",
    "f = open('./Clusters_-_Symbols.tsv')\n",
    "tsv_file = csv.reader(f, delimiter=\"\\t\")\n",
    "\n",
    "# biomart_server\n",
    "server = BiomartServer('http://www.ensembl.org/biomart')\n",
    "dataset = server.datasets['mmusculus_gene_ensembl']\n",
    "\n",
    "\n",
    "# separation_of_headers  \n",
    "headers = next(tsv_file)               \n",
    "rest2 = list(tsv_file)\n",
    "\n",
    "# separation_of_regulation\n",
    "downTab = [row[16] for row in rest2]\n",
    "upTab = [row[17] for row in rest2]\n",
    "\n",
    "# separation_of_rest\n",
    "[row.pop(17) for row in rest2]   \n",
    "[row.pop(16) for row in rest2]\n",
    "\n",
    "rest = []\n",
    "for row in rest2:\n",
    "    for i in range(len(row)):\n",
    "        if row[i] not in rest:\n",
    "            rest.append(row[i])\n",
    "            \n",
    "#print(rest2)\n",
    "#print(rest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9353a0b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Index': 1, 'Gene_name': 'Mef2c', 'Gene_list_id': 'marpiech_tissues_dex_1', 'Ensembl_gene_id': 'ENSMUSG00000005583', 'Ensembl_transcript_id': 'ENSMUST00000198916|ENSMUST00000197022|ENSMUST00000197722|ENSMUST00000199262|ENSMUST00000198360|ENSMUST00000199167|ENSMUST00000198217|ENSMUST00000199210|ENSMUST00000197146|ENSMUST00000200123|ENSMUST00000197938|ENSMUST00000196493|ENSMUST00000200394|ENSMUST00000199105|ENSMUST00000163888|ENSMUST00000196207|ENSMUST00000195904|ENSMUST00000198199|ENSMUST00000195984|ENSMUST00000199450|ENSMUST00000199432|ENSMUST00000196832|ENSMUST00000200138|ENSMUST00000198069|ENSMUST00000199019|ENSMUST00000185052|ENSMUST00000005722|ENSMUST00000197681|ENSMUST00000198064|ENSMUST00000197145|ENSMUST00000196730', 'RefSeq_mRNA_id': 'NM_001347581|NM_001347580|NM_001347576|NM_001347566|NM_001347574|NM_001347573|NM_001347568|NM_001347575|NM_001347579|NM_025282|NM_001347571|NM_001347564|NM_001347572|NM_001347578|NM_001347569|NM_001347567|NM_001170537|NM_001347577', 'HGNC_symbol': 'MEF2C', 'UP': '1', 'DOWN': '1', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 2, 'Gene_name': 'Cxcl1', 'Gene_list_id': 'marpiech_tissues_dex_2', 'Ensembl_gene_id': 'ENSMUSG00000029380', 'Ensembl_transcript_id': 'ENSMUST00000201245|ENSMUST00000031327', 'RefSeq_mRNA_id': 'NM_008176', 'HGNC_symbol': '', 'UP': '2', 'DOWN': '2', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 3, 'Gene_name': 'Id1', 'Gene_list_id': 'marpiech_tissues_dex_3', 'Ensembl_gene_id': 'ENSMUSG00000042745', 'Ensembl_transcript_id': 'ENSMUST00000038368|ENSMUST00000109824', 'RefSeq_mRNA_id': 'NM_010495|NM_001355113|NM_001369018', 'HGNC_symbol': 'ID1', 'UP': '3', 'DOWN': '3', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 4, 'Gene_name': 'Aplnr', 'Gene_list_id': 'marpiech_tissues_dex_4', 'Ensembl_gene_id': 'ENSMUSG00000044338', 'Ensembl_transcript_id': 'ENSMUST00000057019|ENSMUST00000184728', 'RefSeq_mRNA_id': 'NM_011784', 'HGNC_symbol': 'APLNR', 'UP': '4', 'DOWN': '4', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 5, 'Gene_name': 'Phlda2', 'Gene_list_id': 'marpiech_tissues_dex_5', 'Ensembl_gene_id': 'ENSMUSG00000010760', 'Ensembl_transcript_id': 'ENSMUST00000207425|ENSMUST00000207561|ENSMUST00000010904', 'RefSeq_mRNA_id': 'NM_009434', 'HGNC_symbol': 'PHLDA2', 'UP': '5', 'DOWN': '5', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 6, 'Gene_name': 'Stac2', 'Gene_list_id': 'marpiech_tissues_dex_6', 'Ensembl_gene_id': 'ENSMUSG00000017400', 'Ensembl_transcript_id': 'ENSMUST00000017544|ENSMUST00000131519', 'RefSeq_mRNA_id': 'NM_146028', 'HGNC_symbol': 'STAC2', 'UP': '6', 'DOWN': '6', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 7, 'Gene_name': 'Thrsp', 'Gene_list_id': 'marpiech_tissues_dex_7', 'Ensembl_gene_id': 'ENSMUSG00000035686', 'Ensembl_transcript_id': 'ENSMUST00000043077|ENSMUST00000206613', 'RefSeq_mRNA_id': 'NM_009381', 'HGNC_symbol': 'THRSP', 'UP': '7', 'DOWN': '7', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 8, 'Gene_name': 'Neurl1a', 'Gene_list_id': 'marpiech_tissues_dex_8', 'Ensembl_gene_id': 'ENSMUSG00000006435', 'Ensembl_transcript_id': 'ENSMUST00000235290|ENSMUST00000235326|ENSMUST00000111807|ENSMUST00000111808', 'RefSeq_mRNA_id': 'NM_021360|NM_001163480', 'HGNC_symbol': 'NEURL1', 'UP': '8', 'DOWN': '8', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 9, 'Gene_name': 'Serpina1b', 'Gene_list_id': 'marpiech_tissues_dex_9', 'Ensembl_gene_id': 'ENSMUSG00000071178', 'Ensembl_transcript_id': 'ENSMUST00000187220|ENSMUST00000164454|ENSMUST00000095450|ENSMUST00000186166', 'RefSeq_mRNA_id': 'NM_009244', 'HGNC_symbol': 'SERPINA1', 'UP': '9', 'DOWN': '9', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 10, 'Gene_name': 'Pnmt', 'Gene_list_id': 'marpiech_tissues_dex_10', 'Ensembl_gene_id': 'ENSMUSG00000038216', 'Ensembl_transcript_id': 'ENSMUST00000041301|ENSMUST00000124527', 'RefSeq_mRNA_id': 'NM_008890', 'HGNC_symbol': 'PNMT', 'UP': '10', 'DOWN': '10', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 11, 'Gene_name': 'Agrp', 'Gene_list_id': 'marpiech_tissues_dex_11', 'Ensembl_gene_id': 'ENSMUSG00000005705', 'Ensembl_transcript_id': 'ENSMUST00000194654|ENSMUST00000005849|ENSMUST00000194091', 'RefSeq_mRNA_id': 'NM_001271806|NM_007427', 'HGNC_symbol': 'AGRP', 'UP': '11', 'DOWN': '11', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 12, 'Gene_name': 'Gadd45b', 'Gene_list_id': 'marpiech_tissues_dex_12', 'Ensembl_gene_id': 'ENSMUSG00000015312', 'Ensembl_transcript_id': 'ENSMUST00000220246|ENSMUST00000219449|ENSMUST00000015456', 'RefSeq_mRNA_id': 'NM_008655', 'HGNC_symbol': 'GADD45B', 'UP': '12', 'DOWN': '12', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 13, 'Gene_name': 'Farp2', 'Gene_list_id': 'marpiech_tissues_dex_13', 'Ensembl_gene_id': 'ENSMUSG00000034066', 'Ensembl_transcript_id': 'ENSMUST00000120301|ENSMUST00000122402', 'RefSeq_mRNA_id': 'NM_145519', 'HGNC_symbol': 'FARP2', 'UP': '13', 'DOWN': '13', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 14, 'Gene_name': '6620401C13Rik', 'Gene_list_id': 'marpiech_tissues_dex_14', 'Ensembl_gene_id': '', 'Ensembl_transcript_id': '', 'RefSeq_mRNA_id': '', 'HGNC_symbol': '', 'UP': '14', 'DOWN': '14', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 15, 'Gene_name': 'Irak2', 'Gene_list_id': 'marpiech_tissues_dex_15', 'Ensembl_gene_id': 'ENSMUSG00000060477', 'Ensembl_transcript_id': 'ENSMUST00000089023|ENSMUST00000204091|ENSMUST00000155554|ENSMUST00000204744|ENSMUST00000089022|ENSMUST00000143948|ENSMUST00000113024|ENSMUST00000059286|ENSMUST00000203381', 'RefSeq_mRNA_id': 'NM_172161|NM_001113553', 'HGNC_symbol': 'IRAK2', 'UP': '15', 'DOWN': '15', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 16, 'Gene_name': '1500009L16Rik', 'Gene_list_id': 'marpiech_tissues_dex_16', 'Ensembl_gene_id': 'ENSMUSG00000087651', 'Ensembl_transcript_id': 'ENSMUST00000218623|ENSMUST00000150459|ENSMUST00000218196', 'RefSeq_mRNA_id': 'NM_001145198', 'HGNC_symbol': 'C12orf75', 'UP': '16', 'DOWN': '16', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 17, 'Gene_name': '9330132O05Rik', 'Gene_list_id': 'marpiech_tissues_dex_1', 'Ensembl_gene_id': '', 'Ensembl_transcript_id': '', 'RefSeq_mRNA_id': '', 'HGNC_symbol': '', 'UP': '1', 'DOWN': '1', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 18, 'Gene_name': 'Socs3', 'Gene_list_id': 'marpiech_tissues_dex_2', 'Ensembl_gene_id': 'ENSMUSG00000053113', 'Ensembl_transcript_id': 'ENSMUST00000132295|ENSMUST00000123255|ENSMUST00000054002', 'RefSeq_mRNA_id': 'NM_007707', 'HGNC_symbol': 'SOCS3', 'UP': '2', 'DOWN': '2', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 19, 'Gene_name': 'Rasl12', 'Gene_list_id': 'marpiech_tissues_dex_3', 'Ensembl_gene_id': 'ENSMUSG00000041696', 'Ensembl_transcript_id': 'ENSMUST00000215605|ENSMUST00000217172|ENSMUST00000165682|ENSMUST00000085453', 'RefSeq_mRNA_id': 'NM_001162923|NM_001033158', 'HGNC_symbol': 'RASL12', 'UP': '3', 'DOWN': '3', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 20, 'Gene_name': 'Tnfsf10', 'Gene_list_id': 'marpiech_tissues_dex_4', 'Ensembl_gene_id': 'ENSMUSG00000039304', 'Ensembl_transcript_id': 'ENSMUST00000046383|ENSMUST00000174840', 'RefSeq_mRNA_id': 'NM_009425', 'HGNC_symbol': 'TNFSF10', 'UP': '4', 'DOWN': '4', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 21, 'Gene_name': 'Sucnr1', 'Gene_list_id': 'marpiech_tissues_dex_5', 'Ensembl_gene_id': 'ENSMUSG00000027762', 'Ensembl_transcript_id': 'ENSMUST00000029326|ENSMUST00000195544', 'RefSeq_mRNA_id': 'NM_032400', 'HGNC_symbol': 'SUCNR1', 'UP': '5', 'DOWN': '5', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 22, 'Gene_name': 'Cldn4', 'Gene_list_id': 'marpiech_tissues_dex_6', 'Ensembl_gene_id': 'ENSMUSG00000047501', 'Ensembl_transcript_id': 'ENSMUST00000051401', 'RefSeq_mRNA_id': 'NM_009903', 'HGNC_symbol': '', 'UP': '6', 'DOWN': '6', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 23, 'Gene_name': 'Pex11a', 'Gene_list_id': 'marpiech_tissues_dex_7', 'Ensembl_gene_id': 'ENSMUSG00000030545', 'Ensembl_transcript_id': 'ENSMUST00000205297|ENSMUST00000139620|ENSMUST00000150606|ENSMUST00000032761', 'RefSeq_mRNA_id': 'NM_011068', 'HGNC_symbol': 'PEX11A', 'UP': '7', 'DOWN': '7', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 24, 'Gene_name': 'Serpine2', 'Gene_list_id': 'marpiech_tissues_dex_8', 'Ensembl_gene_id': 'ENSMUSG00000026249', 'Ensembl_transcript_id': 'ENSMUST00000189793|ENSMUST00000191026|ENSMUST00000153862|ENSMUST00000191529|ENSMUST00000027467|ENSMUST00000190724', 'RefSeq_mRNA_id': 'NM_009255', 'HGNC_symbol': 'SERPINE2', 'UP': '8', 'DOWN': '8', 'Alias': '', 'Info': 'Up'}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Index': 25, 'Gene_name': '9530053J19Rik', 'Gene_list_id': 'marpiech_tissues_dex_9', 'Ensembl_gene_id': '', 'Ensembl_transcript_id': '', 'RefSeq_mRNA_id': '', 'HGNC_symbol': '', 'UP': '9', 'DOWN': '9', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 26, 'Gene_name': 'Bmp4', 'Gene_list_id': 'marpiech_tissues_dex_10', 'Ensembl_gene_id': 'ENSMUSG00000021835', 'Ensembl_transcript_id': 'ENSMUST00000135408|ENSMUST00000074077|ENSMUST00000100676|ENSMUST00000226759', 'RefSeq_mRNA_id': 'NM_007554|NM_001316360', 'HGNC_symbol': 'BMP4', 'UP': '10', 'DOWN': '10', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 27, 'Gene_name': 'Tom1l2', 'Gene_list_id': 'marpiech_tissues_dex_11', 'Ensembl_gene_id': 'ENSMUSG00000000538', 'Ensembl_transcript_id': 'ENSMUST00000064019|ENSMUST00000143124|ENSMUST00000095254|ENSMUST00000151284|ENSMUST00000093046|ENSMUST00000133420|ENSMUST00000102682|ENSMUST00000153920|ENSMUST00000102683|ENSMUST00000093048|ENSMUST00000142225', 'RefSeq_mRNA_id': 'NM_001039092|NM_153080|NM_001403895|NM_001403896|NM_001039093|NM_001359810|NM_001403893', 'HGNC_symbol': 'TOM1L2', 'UP': '11', 'DOWN': '11', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 28, 'Gene_name': 'Ppp1r3g', 'Gene_list_id': 'marpiech_tissues_dex_12', 'Ensembl_gene_id': 'ENSMUSG00000050423', 'Ensembl_transcript_id': 'ENSMUST00000172766|ENSMUST00000225537|ENSMUST00000132661', 'RefSeq_mRNA_id': 'NM_029628', 'HGNC_symbol': 'PPP1R3G', 'UP': '12', 'DOWN': '12', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 29, 'Gene_name': 'Map2k3os', 'Gene_list_id': 'marpiech_tissues_dex_13', 'Ensembl_gene_id': 'ENSMUSG00000042549', 'Ensembl_transcript_id': 'ENSMUST00000123544|ENSMUST00000243473|ENSMUST00000243488', 'RefSeq_mRNA_id': '', 'HGNC_symbol': '', 'UP': '13', 'DOWN': '13', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 30, 'Gene_name': 'Arrdc3', 'Gene_list_id': 'marpiech_tissues_dex_14', 'Ensembl_gene_id': 'ENSMUSG00000074794', 'Ensembl_transcript_id': 'ENSMUST00000159690|ENSMUST00000161441|ENSMUST00000162904|ENSMUST00000159856|ENSMUST00000159090|ENSMUST00000099356', 'RefSeq_mRNA_id': 'NM_001042591', 'HGNC_symbol': 'ARRDC3', 'UP': '14', 'DOWN': '14', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 31, 'Gene_name': 'Chac1', 'Gene_list_id': 'marpiech_tissues_dex_15', 'Ensembl_gene_id': 'ENSMUSG00000027313', 'Ensembl_transcript_id': 'ENSMUST00000028780', 'RefSeq_mRNA_id': 'NM_026929', 'HGNC_symbol': 'CHAC1', 'UP': '15', 'DOWN': '15', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 32, 'Gene_name': 'Ccl7', 'Gene_list_id': 'marpiech_tissues_dex_16', 'Ensembl_gene_id': 'ENSMUSG00000035373', 'Ensembl_transcript_id': 'ENSMUST00000021011', 'RefSeq_mRNA_id': 'NM_013654', 'HGNC_symbol': 'CCL7', 'UP': '16', 'DOWN': '16', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 33, 'Gene_name': 'Gpr81', 'Gene_list_id': 'marpiech_tissues_dex_1', 'Ensembl_gene_id': '', 'Ensembl_transcript_id': '', 'RefSeq_mRNA_id': '', 'HGNC_symbol': '', 'UP': '1', 'DOWN': '1', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 34, 'Gene_name': 'Sox7', 'Gene_list_id': 'marpiech_tissues_dex_2', 'Ensembl_gene_id': 'ENSMUSG00000063060', 'Ensembl_transcript_id': 'ENSMUST00000079652', 'RefSeq_mRNA_id': 'NM_011446', 'HGNC_symbol': 'SOX7', 'UP': '2', 'DOWN': '2', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 35, 'Gene_name': 'Sap30', 'Gene_list_id': 'marpiech_tissues_dex_3', 'Ensembl_gene_id': 'ENSMUSG00000031609', 'Ensembl_transcript_id': 'ENSMUST00000034022', 'RefSeq_mRNA_id': 'NM_021788', 'HGNC_symbol': 'SAP30', 'UP': '3', 'DOWN': '3', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 36, 'Gene_name': 'C2cd4a', 'Gene_list_id': 'marpiech_tissues_dex_4', 'Ensembl_gene_id': 'ENSMUSG00000047990', 'Ensembl_transcript_id': 'ENSMUST00000054500', 'RefSeq_mRNA_id': 'NM_001163143', 'HGNC_symbol': '', 'UP': '4', 'DOWN': '4', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 37, 'Gene_name': 'Atp1b1', 'Gene_list_id': 'marpiech_tissues_dex_5', 'Ensembl_gene_id': 'ENSMUSG00000026576', 'Ensembl_transcript_id': 'ENSMUST00000193367|ENSMUST00000193980|ENSMUST00000192522|ENSMUST00000027863', 'RefSeq_mRNA_id': 'NM_009721', 'HGNC_symbol': 'ATP1B1', 'UP': '5', 'DOWN': '5', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 38, 'Gene_name': 'Mup1', 'Gene_list_id': 'marpiech_tissues_dex_6', 'Ensembl_gene_id': 'ENSMUSG00000078683', 'Ensembl_transcript_id': 'ENSMUST00000107499|ENSMUST00000135953|ENSMUST00000084548|ENSMUST00000103012', 'RefSeq_mRNA_id': 'NM_001163011|NM_001163010|NM_031188', 'HGNC_symbol': '', 'UP': '6', 'DOWN': '6', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 39, 'Gene_name': 'Ppp1r3c', 'Gene_list_id': 'marpiech_tissues_dex_7', 'Ensembl_gene_id': 'ENSMUSG00000067279', 'Ensembl_transcript_id': 'ENSMUST00000087321', 'RefSeq_mRNA_id': 'NM_016854', 'HGNC_symbol': 'PPP1R3C', 'UP': '7', 'DOWN': '7', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 40, 'Gene_name': 'Sla', 'Gene_list_id': 'marpiech_tissues_dex_8', 'Ensembl_gene_id': 'ENSMUSG00000022372', 'Ensembl_transcript_id': 'ENSMUST00000168589|ENSMUST00000164163|ENSMUST00000168522|ENSMUST00000100572|ENSMUST00000164939', 'RefSeq_mRNA_id': 'NM_001029841|NM_009192', 'HGNC_symbol': 'SLA', 'UP': '8', 'DOWN': '8', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 41, 'Gene_name': 'Cyp2b10', 'Gene_list_id': 'marpiech_tissues_dex_9', 'Ensembl_gene_id': 'ENSMUSG00000030483', 'Ensembl_transcript_id': 'ENSMUST00000144140|ENSMUST00000005477|ENSMUST00000072438', 'RefSeq_mRNA_id': 'NM_009999', 'HGNC_symbol': '', 'UP': '9', 'DOWN': '9', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 42, 'Gene_name': 'Plekhf1', 'Gene_list_id': 'marpiech_tissues_dex_10', 'Ensembl_gene_id': 'ENSMUSG00000074170', 'Ensembl_transcript_id': 'ENSMUST00000098513', 'RefSeq_mRNA_id': 'NM_024413', 'HGNC_symbol': 'PLEKHF1', 'UP': '10', 'DOWN': '10', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 43, 'Gene_name': 'Kcnf1', 'Gene_list_id': 'marpiech_tissues_dex_11', 'Ensembl_gene_id': 'ENSMUSG00000051726', 'Ensembl_transcript_id': 'ENSMUST00000170580', 'RefSeq_mRNA_id': 'NM_201531', 'HGNC_symbol': 'KCNF1', 'UP': '11', 'DOWN': '11', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 44, 'Gene_name': 'Znfx1', 'Gene_list_id': 'marpiech_tissues_dex_12', 'Ensembl_gene_id': 'ENSMUSG00000039501', 'Ensembl_transcript_id': 'ENSMUST00000048988|ENSMUST00000128676|ENSMUST00000067584|ENSMUST00000135967|ENSMUST00000155281', 'RefSeq_mRNA_id': 'NM_001033196|NM_001291162', 'HGNC_symbol': 'ZNFX1', 'UP': '12', 'DOWN': '12', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 45, 'Gene_name': 'Dusp18', 'Gene_list_id': 'marpiech_tissues_dex_13', 'Ensembl_gene_id': 'ENSMUSG00000047205', 'Ensembl_transcript_id': 'ENSMUST00000055931|ENSMUST00000109996', 'RefSeq_mRNA_id': 'NM_173745', 'HGNC_symbol': 'DUSP18', 'UP': '13', 'DOWN': '13', 'Alias': '', 'Info': 'Up'}\n",
      "{'Index': 46, 'Gene_name': 'Klk1b22', 'Gene_list_id': 'marpiech_tissues_dex_14', 'Ensembl_gene_id': 'ENSMUSG00000060177', 'Ensembl_transcript_id': 'ENSMUST00000077528', 'RefSeq_mRNA_id': 'NM_010114', 'HGNC_symbol': 'KLK3', 'UP': '14', 'DOWN': '14', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 47, 'Gene_name': 'Irs3', 'Gene_list_id': 'marpiech_tissues_dex_15', 'Ensembl_gene_id': 'ENSMUSG00000047182', 'Ensembl_transcript_id': 'ENSMUST00000196511|ENSMUST00000057314', 'RefSeq_mRNA_id': 'NM_010571', 'HGNC_symbol': '', 'UP': '15', 'DOWN': '15', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 48, 'Gene_name': 'Plekho2', 'Gene_list_id': 'marpiech_tissues_dex_16', 'Ensembl_gene_id': 'ENSMUSG00000050721', 'Ensembl_transcript_id': 'ENSMUST00000213652|ENSMUST00000214740|ENSMUST00000068944|ENSMUST00000213554', 'RefSeq_mRNA_id': 'NM_153119', 'HGNC_symbol': 'PLEKHO2', 'UP': '16', 'DOWN': '16', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 49, 'Gene_name': 'Arhgef39', 'Gene_list_id': 'marpiech_tissues_dex_1', 'Ensembl_gene_id': 'ENSMUSG00000051517', 'Ensembl_transcript_id': 'ENSMUST00000147124|ENSMUST00000136005|ENSMUST00000154905|ENSMUST00000146781|ENSMUST00000125834|ENSMUST00000152134|ENSMUST00000156058|ENSMUST00000128754|ENSMUST00000054538|ENSMUST00000129378', 'RefSeq_mRNA_id': 'NM_001013377', 'HGNC_symbol': 'ARHGEF39', 'UP': '1', 'DOWN': '1', 'Alias': '', 'Info': 'Down'}\n",
      "{'Index': 50, 'Gene_name': 'Sstr2', 'Gene_list_id': 'marpiech_tissues_dex_2', 'Ensembl_gene_id': 'ENSMUSG00000047904', 'Ensembl_transcript_id': 'ENSMUST00000146390|ENSMUST00000067591|ENSMUST00000106630', 'RefSeq_mRNA_id': 'NM_009217|NM_001042606', 'HGNC_symbol': 'SSTR2', 'UP': '2', 'DOWN': '2', 'Alias': '', 'Info': 'Down'}\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_9952\\3219890050.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     44\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     45\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 46\u001b[1;33m         \u001b[0mls_biomartHumanOrthologs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbiomartHumanOrthologs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene_name\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     47\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mls_biomartHumanOrthologs\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     48\u001b[0m             \u001b[0mhgnc_symbol\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m''\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_9952\\668787817.py\u001b[0m in \u001b[0;36mbiomartHumanOrthologs\u001b[1;34m(mgi_symbol)\u001b[0m\n\u001b[0;32m     68\u001b[0m     \u001b[0mattributes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m'hsapiens_homolog_associated_gene_name'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     69\u001b[0m     \u001b[0mfilters\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;34m'mgi_symbol'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mmgi_symbol\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 70\u001b[1;33m     \u001b[0mresponse\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdataset\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msearch\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'attributes'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mattributes\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'filters'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mfilters\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     71\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     72\u001b[0m \u001b[1;31m# response_convertion\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\biomart\\dataset.py\u001b[0m in \u001b[0;36msearch\u001b[1;34m(self, params, header, count)\u001b[0m\n\u001b[0;32m    180\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    181\u001b[0m             \u001b[1;31m# discover attributes and pages\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 182\u001b[1;33m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfetch_attributes\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    183\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    184\u001b[0m             \u001b[1;31m# no attributes given, use default attributes\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\biomart\\dataset.py\u001b[0m in \u001b[0;36mfetch_attributes\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    135\u001b[0m         \u001b[1;31m# grab attribute details\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    136\u001b[0m         \u001b[0mr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mserver\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_request\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"attributes\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdataset\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 137\u001b[1;33m         \u001b[1;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0miter_lines\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    138\u001b[0m             \u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mline\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdecode\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'utf8'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    139\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mline\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\requests\\models.py\u001b[0m in \u001b[0;36miter_lines\u001b[1;34m(self, chunk_size, decode_unicode, delimiter)\u001b[0m\n\u001b[0;32m    863\u001b[0m         \u001b[0mpending\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    864\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 865\u001b[1;33m         for chunk in self.iter_content(\n\u001b[0m\u001b[0;32m    866\u001b[0m             \u001b[0mchunk_size\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mchunk_size\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdecode_unicode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdecode_unicode\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    867\u001b[0m         ):\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\requests\\models.py\u001b[0m in \u001b[0;36mgenerate\u001b[1;34m()\u001b[0m\n\u001b[0;32m    814\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mhasattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mraw\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"stream\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    815\u001b[0m                 \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 816\u001b[1;33m                     \u001b[1;32myield\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mraw\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstream\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mchunk_size\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdecode_content\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    817\u001b[0m                 \u001b[1;32mexcept\u001b[0m \u001b[0mProtocolError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    818\u001b[0m                     \u001b[1;32mraise\u001b[0m \u001b[0mChunkedEncodingError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\urllib3\\response.py\u001b[0m in \u001b[0;36mstream\u001b[1;34m(self, amt, decode_content)\u001b[0m\n\u001b[0;32m    621\u001b[0m         \"\"\"\n\u001b[0;32m    622\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchunked\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msupports_chunked_reads\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 623\u001b[1;33m             \u001b[1;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread_chunked\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdecode_content\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdecode_content\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    624\u001b[0m                 \u001b[1;32myield\u001b[0m \u001b[0mline\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    625\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\urllib3\\response.py\u001b[0m in \u001b[0;36mread_chunked\u001b[1;34m(self, amt, decode_content)\u001b[0m\n\u001b[0;32m    816\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchunk_left\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    817\u001b[0m                     \u001b[1;32mbreak\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 818\u001b[1;33m                 \u001b[0mchunk\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_handle_chunk\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    819\u001b[0m                 decoded = self._decode(\n\u001b[0;32m    820\u001b[0m                     \u001b[0mchunk\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdecode_content\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdecode_content\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mflush_decoder\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\urllib3\\response.py\u001b[0m in \u001b[0;36m_handle_chunk\u001b[1;34m(self, amt)\u001b[0m\n\u001b[0;32m    760\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchunk_left\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    761\u001b[0m         \u001b[1;32melif\u001b[0m \u001b[0mamt\u001b[0m \u001b[1;33m<\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchunk_left\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 762\u001b[1;33m             \u001b[0mvalue\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_fp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_safe_read\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    763\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchunk_left\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchunk_left\u001b[0m \u001b[1;33m-\u001b[0m \u001b[0mamt\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    764\u001b[0m             \u001b[0mreturned_chunk\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_safe_read\u001b[1;34m(self, amt)\u001b[0m\n\u001b[0;32m    624\u001b[0m         \u001b[0ms\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    625\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[0mamt\u001b[0m \u001b[1;33m>\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 626\u001b[1;33m             \u001b[0mchunk\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mMAXAMOUNT\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    627\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mchunk\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    628\u001b[0m                 \u001b[1;32mraise\u001b[0m \u001b[0mIncompleteRead\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mb''\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mamt\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[1;34m(self, b)\u001b[0m\n\u001b[0;32m    702\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    703\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 704\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    705\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    706\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# CREATING FINAL LIST\n",
    "ls_geneDictionaries = []\n",
    "ls_biomartParameters = []\n",
    "ls_biomartHumanOrthologs = []\n",
    "ls_notResponse = []\n",
    "\n",
    "\n",
    "for i in range(len(rest)):\n",
    "    gene_name = rest[i]\n",
    "#   print(gene_name)\n",
    "    if gene_name != '':\n",
    "\n",
    "        # variables\n",
    "\n",
    "        geneSource = 'marpiech_tissues_dex'\n",
    "        cluster = headers[i % 16]\n",
    "        gene_list_id = geneSource + '_' + cluster\n",
    "        index = i + 1\n",
    "        direction = regulation(gene_name)          # => info\n",
    "        alias = ''\n",
    "\n",
    "\n",
    "        ensembl_gene_id_temp = []\n",
    "        ensembl_transcript_id_temp = []\n",
    "        refseq_mrna_temp = []\n",
    "\n",
    "        ls_biomartParameters = biomartParameters(gene_name)\n",
    "        if not ls_biomartParameters:\n",
    "            ls_notResponse.append(gene_name)\n",
    "#            print(gene_name)\n",
    "\n",
    "\n",
    "        for ls in ls_biomartParameters:\n",
    "            for j in range(3):\n",
    "                if len(ls) < (j+1):\n",
    "                    ls.append('')\n",
    "            ensembl_gene_id_temp.append(ls[0])\n",
    "            ensembl_transcript_id_temp.append(ls[1])\n",
    "            refseq_mrna_temp.append(ls[2])  \n",
    "\n",
    "        ensembl_gene_id = '|'.join(list(set(filter(None, ensembl_gene_id_temp))))\n",
    "        ensembl_transcript_id = '|'.join(list(set(filter(None, ensembl_transcript_id_temp))))\n",
    "        refseq_mrna = '|'.join(list(set(filter(None, refseq_mrna_temp))))\n",
    "                               \n",
    "\n",
    "        ls_biomartHumanOrthologs = biomartHumanOrthologs(gene_name)\n",
    "        if not ls_biomartHumanOrthologs:\n",
    "            hgnc_symbol = ''\n",
    "        else:\n",
    "            hgnc_symbol = ls_biomartHumanOrthologs[0][0]\n",
    "        \n",
    "\n",
    "\n",
    "        temp_gene_dictionary = gene_dictionary(index,\n",
    "                                               gene_name,\n",
    "                                               cluster,\n",
    "                                               gene_list_id,\n",
    "                                               ensembl_gene_id,\n",
    "                                               ensembl_transcript_id, \n",
    "                                               refseq_mrna,\n",
    "                                               hgnc_symbol,\n",
    "                                               alias,\n",
    "                                               direction)\n",
    "\n",
    "        ls_geneDictionaries.append(temp_gene_dictionary)\n",
    "        print(temp_gene_dictionary)\n",
    "\n",
    "data = ls_geneDictionaries\n",
    "\n",
    "df = pd.DataFrame.from_dict(data)\n",
    "\n",
    "\n",
    "\n",
    "\"\"\"\n",
    "for gene in ls_geneDictionaries:\n",
    "print(gene)\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "b3ffb1e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['MGI:95292', 'MGI:95292', 'MGI:95537', 'MGI:106023', 'MGI:107565', 'MGI:109382', 'MGI:88613', 'MGI:1277133', 'MGI:1314653', 'MGI:1858441', 'MGI:1860303', 'MGI:1914701', 'MGI:1915368', 'MGI:1915982', 'MGI:1916318', 'MGI:1918730', 'MGI:1921298', 'MGI:1921765', 'MGI:1922263', 'MGI:1924234', 'MGI:1924802', 'MGI:1924879', 'MGI:1924967', 'MGI:1933383', 'MGI:2441671', 'MGI:2442377', 'MGI:2442733', 'MGI:2442910', 'MGI:2443629', 'MGI:2444394', 'MGI:2670959', 'MGI:2685484', 'MGI:3643534', 'MGI:3645360', 'MGI:3646928', 'MGI:3648717']\n"
     ]
    }
   ],
   "source": [
    "import openpyxl \n",
    "\n",
    "\n",
    "wb = openpyxl.load_workbook(r'C:\\Users\\joasi\\geneBase\\1_pub\\Wyniki\\response_1_pub_200323.xlsx')\n",
    "ws = wb.active\n",
    "\n",
    "mgi_id = [ws.cell(row=i,column=2).value for i in range(2,ws.max_row+1)]\n",
    "\n",
    "\n",
    "gene_name_2 = [ws.cell(row=i,column=1).value for i in range(2,ws.max_row+1)]\n",
    "\n",
    "\n",
    "print(mgi_id)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90717532",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MGI:95292\n",
      "[]\n",
      "   \n",
      "MGI:95292\n",
      "[]\n",
      "   \n",
      "MGI:95537\n",
      "[]\n",
      "   \n",
      "MGI:106023\n",
      "[]\n",
      "   \n",
      "MGI:107565\n",
      "[]\n",
      "   \n",
      "MGI:109382\n",
      "[]\n",
      "   \n",
      "MGI:88613\n",
      "[]\n",
      "   \n",
      "MGI:1277133\n",
      "[]\n",
      "   \n",
      "MGI:1314653\n",
      "[]\n",
      "   \n",
      "MGI:1858441\n",
      "[]\n",
      "   \n",
      "MGI:1860303\n"
     ]
    }
   ],
   "source": [
    "# SECOND RESPONSE\n",
    "server = BiomartServer('http://www.ensembl.org/biomart')\n",
    "dataset = server.datasets['mmusculus_gene_ensembl']\n",
    "\n",
    "def biomartParameters_mgi(mgi_id):\n",
    "\n",
    "    attributes = ['ensembl_gene_id',\n",
    "                  'ensembl_transcript_id',\n",
    "                  'refseq_mrna']\n",
    "    print(mgi_id)\n",
    "    filters = {'mgi_id':[mgi_id]}                     # gene_name = mgi_symbol\n",
    "    response = dataset.search({'attributes':attributes,'filters':filters})\n",
    "    \n",
    "\n",
    "# response_convertion\n",
    "    lines = response.text.split(\"\\n\")\n",
    "    lines = list(filter(lambda l: l != \"\",lines))\n",
    "    values = []\n",
    "    for l in lines:\n",
    "        temp = []\n",
    "        raw_values = l.split(\"\\t\")\n",
    "        values.append(raw_values)\n",
    "        \n",
    "# if values not return ????\n",
    "\n",
    "    return values  \n",
    "\n",
    "def biomartHumanOrthologs_mgi(mgi_id):\n",
    "    attributes = ['hsapiens_homolog_associated_gene_name']\n",
    "    \n",
    "    filters = {'mgi_id':[mgi_id]}\n",
    "    response = dataset.search({'attributes':attributes,'filters':filters})\n",
    "    \n",
    "# response_convertion\n",
    "    lines = response.text.split(\"\\n\")\n",
    "    lines = list(filter(lambda l: l != \"\",lines))\n",
    "    values = []\n",
    "    for l in lines:\n",
    "        temp = []\n",
    "        raw_values = l.split(\"\\t\")\n",
    "        values.append(raw_values)\n",
    "        \n",
    "# if values not return ????\n",
    "\n",
    "    return values\n",
    "\n",
    "ls_notResponse_after = []\n",
    "ls_biomartParameters = []\n",
    "ls_biomartHumanOrthologs = []\n",
    "\n",
    "for i in range(len(mgi_id)):\n",
    "    mgi_ID = mgi_id[i]\n",
    "    gene_name = gene_name_2[i]\n",
    "    print(mgi_ID)\n",
    "    if gene_name != '':\n",
    "\n",
    "        # variables\n",
    "        alias = ''\n",
    "\n",
    "\n",
    "        ensembl_gene_id_temp = []\n",
    "        ensembl_transcript_id_temp = []\n",
    "        refseq_mrna_temp = []\n",
    "\n",
    "        ls_biomartParameters = biomartParameters_mgi(mgi_ID)\n",
    "        \n",
    "        print(ls_biomartParameters)\n",
    "        \n",
    "        if not ls_biomartParameters:\n",
    "            ls_notResponse_after.append(mgi_ID)\n",
    "#            print(gene_name)\n",
    "\n",
    "\n",
    "        for ls in ls_biomartParameters:\n",
    "            for j in range(3):\n",
    "                if len(ls) < (j+1):\n",
    "                    ls.append('')\n",
    "            ensembl_gene_id_temp.append(ls[0])\n",
    "            ensembl_transcript_id_temp.append(ls[1])\n",
    "            refseq_mrna_temp.append(ls[2])  \n",
    "\n",
    "        ensembl_gene_id = '|'.join(list(set(filter(None, ensembl_gene_id_temp))))\n",
    "        ensembl_transcript_id = '|'.join(list(set(filter(None, ensembl_transcript_id_temp))))\n",
    "        refseq_mrna = '|'.join(list(set(filter(None, refseq_mrna_temp))))\n",
    "                               \n",
    "\n",
    "        ls_biomartHumanOrthologs = biomartHumanOrthologs_mgi(mgi_ID)\n",
    "        if not ls_biomartHumanOrthologs:\n",
    "            hgnc_symbol = ''\n",
    "        else:\n",
    "            hgnc_symbol = ls_biomartHumanOrthologs[0][0]\n",
    "        print(ensembl_gene_id, ensembl_transcript_id, refseq_mrna, hgnc_symbol)\n",
    "        for k in range(len(ls_geneDictionaries)):\n",
    "            if ls_geneDictionaries[k]['Gene_name'] == gene_name:\n",
    "                ls_geneDictionaries[k]['Ensembl_gene_id'] = ensembl_gene_id\n",
    "                ls_geneDictionaries[k]['Ensembl_transcript_id'] = ensembl_transcript_id\n",
    "                ls_geneDictionaries[k]['RefSeq_mRNA_id'] = refseq_mrna\n",
    "                ls_geneDictionaries[k]['HGNC_symbol'] = hgnc_symbol\n",
    "                \n",
    "\n",
    "                \n",
    "data = ls_geneDictionaries\n",
    "df = pd.DataFrame.from_dict(data)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08496e6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4f95c435",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Index</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Gene_list_id</th>\n",
       "      <th>Ensembl_gene_id</th>\n",
       "      <th>Ensembl_transcript_id</th>\n",
       "      <th>RefSeq_mRNA_id</th>\n",
       "      <th>HGNC_symbol</th>\n",
       "      <th>UP</th>\n",
       "      <th>DOWN</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Mef2c</td>\n",
       "      <td>marpiech_tissues_dex_1</td>\n",
       "      <td>ENSMUSG00000005583</td>\n",
       "      <td>ENSMUST00000199262|ENSMUST00000199019|ENSMUST0...</td>\n",
       "      <td>NM_001347581|NM_001347564|NM_001347568|NM_0013...</td>\n",
       "      <td>MEF2C</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Down</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Cxcl1</td>\n",
       "      <td>marpiech_tissues_dex_2</td>\n",
       "      <td>ENSMUSG00000029380</td>\n",
       "      <td>ENSMUST00000201245|ENSMUST00000031327</td>\n",
       "      <td>NM_008176</td>\n",
       "      <td></td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td>Down</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>Id1</td>\n",
       "      <td>marpiech_tissues_dex_3</td>\n",
       "      <td>ENSMUSG00000042745</td>\n",
       "      <td>ENSMUST00000038368|ENSMUST00000109824</td>\n",
       "      <td>NM_001369018|NM_010495|NM_001355113</td>\n",
       "      <td>ID1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td>Down</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>Aplnr</td>\n",
       "      <td>marpiech_tissues_dex_4</td>\n",
       "      <td>ENSMUSG00000044338</td>\n",
       "      <td>ENSMUST00000057019|ENSMUST00000184728</td>\n",
       "      <td>NM_011784</td>\n",
       "      <td>APLNR</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td>Down</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>Phlda2</td>\n",
       "      <td>marpiech_tissues_dex_5</td>\n",
       "      <td>ENSMUSG00000010760</td>\n",
       "      <td>ENSMUST00000207561|ENSMUST00000207425|ENSMUST0...</td>\n",
       "      <td>NM_009434</td>\n",
       "      <td>PHLDA2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td>Down</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>476</td>\n",
       "      <td>Polr3e</td>\n",
       "      <td>marpiech_tissues_dex_12</td>\n",
       "      <td>ENSMUSG00000030880</td>\n",
       "      <td>ENSMUST00000209014|ENSMUST00000207481|ENSMUST0...</td>\n",
       "      <td>NM_001164096|NM_025298|NM_001379288</td>\n",
       "      <td>POLR3E</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td></td>\n",
       "      <td>Up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>477</td>\n",
       "      <td>Arhgef28</td>\n",
       "      <td>marpiech_tissues_dex_13</td>\n",
       "      <td>ENSMUSG00000021662</td>\n",
       "      <td>ENSMUST00000109426|ENSMUST00000238746|ENSMUST0...</td>\n",
       "      <td>NM_001408685|NM_001408684|NM_012026|NM_001408686</td>\n",
       "      <td>ARHGEF28</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td></td>\n",
       "      <td>Up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>478</td>\n",
       "      <td>Fgfbp1</td>\n",
       "      <td>marpiech_tissues_dex_14</td>\n",
       "      <td>ENSMUSG00000048373</td>\n",
       "      <td>ENSMUST00000199481|ENSMUST00000061299|ENSMUST0...</td>\n",
       "      <td>NM_008009|NM_001271616</td>\n",
       "      <td>FGFBP1</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td>Up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>479</td>\n",
       "      <td>Uap1</td>\n",
       "      <td>marpiech_tissues_dex_15</td>\n",
       "      <td>ENSMUSG00000026670</td>\n",
       "      <td>ENSMUST00000191797|ENSMUST00000191690|ENSMUST0...</td>\n",
       "      <td>NM_133806|NM_001305046|NM_001305045</td>\n",
       "      <td>UAP1</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td></td>\n",
       "      <td>Up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>478</th>\n",
       "      <td>480</td>\n",
       "      <td>Cul3</td>\n",
       "      <td>marpiech_tissues_dex_16</td>\n",
       "      <td>ENSMUSG00000004364</td>\n",
       "      <td>ENSMUST00000168372|ENSMUST00000163119|ENSMUST0...</td>\n",
       "      <td>NM_016716|NM_001313728</td>\n",
       "      <td>CUL3</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td></td>\n",
       "      <td>Up</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>479 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Index Gene_name             Gene_list_id     Ensembl_gene_id  \\\n",
       "0        1     Mef2c   marpiech_tissues_dex_1  ENSMUSG00000005583   \n",
       "1        2     Cxcl1   marpiech_tissues_dex_2  ENSMUSG00000029380   \n",
       "2        3       Id1   marpiech_tissues_dex_3  ENSMUSG00000042745   \n",
       "3        4     Aplnr   marpiech_tissues_dex_4  ENSMUSG00000044338   \n",
       "4        5    Phlda2   marpiech_tissues_dex_5  ENSMUSG00000010760   \n",
       "..     ...       ...                      ...                 ...   \n",
       "474    476    Polr3e  marpiech_tissues_dex_12  ENSMUSG00000030880   \n",
       "475    477  Arhgef28  marpiech_tissues_dex_13  ENSMUSG00000021662   \n",
       "476    478    Fgfbp1  marpiech_tissues_dex_14  ENSMUSG00000048373   \n",
       "477    479      Uap1  marpiech_tissues_dex_15  ENSMUSG00000026670   \n",
       "478    480      Cul3  marpiech_tissues_dex_16  ENSMUSG00000004364   \n",
       "\n",
       "                                 Ensembl_transcript_id  \\\n",
       "0    ENSMUST00000199262|ENSMUST00000199019|ENSMUST0...   \n",
       "1                ENSMUST00000201245|ENSMUST00000031327   \n",
       "2                ENSMUST00000038368|ENSMUST00000109824   \n",
       "3                ENSMUST00000057019|ENSMUST00000184728   \n",
       "4    ENSMUST00000207561|ENSMUST00000207425|ENSMUST0...   \n",
       "..                                                 ...   \n",
       "474  ENSMUST00000209014|ENSMUST00000207481|ENSMUST0...   \n",
       "475  ENSMUST00000109426|ENSMUST00000238746|ENSMUST0...   \n",
       "476  ENSMUST00000199481|ENSMUST00000061299|ENSMUST0...   \n",
       "477  ENSMUST00000191797|ENSMUST00000191690|ENSMUST0...   \n",
       "478  ENSMUST00000168372|ENSMUST00000163119|ENSMUST0...   \n",
       "\n",
       "                                        RefSeq_mRNA_id HGNC_symbol  UP DOWN  \\\n",
       "0    NM_001347581|NM_001347564|NM_001347568|NM_0013...       MEF2C   1    1   \n",
       "1                                            NM_008176               2    2   \n",
       "2                  NM_001369018|NM_010495|NM_001355113         ID1   3    3   \n",
       "3                                            NM_011784       APLNR   4    4   \n",
       "4                                            NM_009434      PHLDA2   5    5   \n",
       "..                                                 ...         ...  ..  ...   \n",
       "474                NM_001164096|NM_025298|NM_001379288      POLR3E  12   12   \n",
       "475   NM_001408685|NM_001408684|NM_012026|NM_001408686    ARHGEF28  13   13   \n",
       "476                             NM_008009|NM_001271616      FGFBP1  14   14   \n",
       "477                NM_133806|NM_001305046|NM_001305045        UAP1  15   15   \n",
       "478                             NM_016716|NM_001313728        CUL3  16   16   \n",
       "\n",
       "    Alias  Info  \n",
       "0          Down  \n",
       "1          Down  \n",
       "2          Down  \n",
       "3          Down  \n",
       "4          Down  \n",
       "..    ...   ...  \n",
       "474          Up  \n",
       "475          Up  \n",
       "476          Up  \n",
       "477          Up  \n",
       "478          Up  \n",
       "\n",
       "[479 rows x 11 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n"
     ]
    }
   ],
   "source": [
    "# WYNIK_DATAFRAME\n",
    "print(display(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "629c3e24",
   "metadata": {},
   "outputs": [],
   "source": [
    "# WYNIK_Excell\n",
    "df.to_excel(r'C:\\Users\\joasi\\geneBase\\1_pub\\Wyniki\\geneBase_probe_2003233.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6c1e4880",
   "metadata": {},
   "outputs": [],
   "source": [
    "# WYNIK_TSV\n",
    "df.to_csv(r'C:\\Users\\joasi\\geneBase\\1_pub\\Wyniki\\geneBase_170323.tsv', sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a08a1c8c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6620401C13Rik\n",
      "9330132O05Rik\n",
      "9530053J19Rik\n",
      "Gpr81\n",
      "A830037N18Rik\n",
      "9430023L20Rik\n",
      "Cyr61\n",
      "Gm6484\n",
      "CKLF5\n",
      "9830132E05Rik\n",
      "Niacr1\n",
      "LOC332300\n",
      "LOC381739\n",
      "LOC100040353\n",
      "LOC672274\n",
      "6030408B16Rik\n",
      "LOC628050\n",
      "3110023G01Rik\n",
      "Nrp\n",
      "Ccrn4l\n",
      "1200012P04Rik\n",
      "A530020H22Rik\n",
      "9930013L23Rik\n",
      "Ctgf\n",
      "EG245297\n",
      "D030063E12\n",
      "9330177P18Rik\n",
      "LOC624610\n",
      "Fam129b\n",
      "1700081H05Rik\n",
      "4930544G21Rik\n",
      "LOC100047902\n",
      "Mllt4\n",
      "Lrrc16\n",
      "Grasp\n",
      "LOC665946\n",
      "Hmga2-ps1\n",
      "BC020108\n",
      "4632428N05Rik\n",
      "Egfbp2\n",
      "4930502E18Rik\n",
      "D14Ertd668e\n",
      "A530026G17\n",
      "Mela\n",
      "BC030863\n",
      "LOC100041932\n",
      "A130038J17Rik\n",
      "1810011O10Rik\n",
      "8430408G22Rik\n",
      "Amica1\n",
      "EG668311\n",
      "EG640530\n",
      "52\n"
     ]
    }
   ],
   "source": [
    "# ls_notResponse => TSV\n",
    "length = 0\n",
    "for ls in ls_notResponse:\n",
    "    print(ls)\n",
    "    length += 1\n",
    "    \n",
    "print(length)    \n",
    "data1 = ls_notResponse\n",
    "df = pd.DataFrame.from_dict(data1)\n",
    "df.to_csv(r'C:\\Users\\joasi\\geneBase\\1_pub\\Wyniki\\ls_notResponse_200323.tsv', sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9099dee1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSMUSG00000023087\tNoct\n",
      "ENSMUSG00000068245\tPhf11d\n",
      "ENSMUSG00000068036\tAfdn\n",
      "ENSMUSG00000000531\tTamalin\n",
      "ENSMUSG00000041957\tPkp2\n",
      "ENSMUSG00000037204\tAtg101\n",
      "ENSMUSG00000021338\tCarmil1\n",
      "ENSMUSG00000056313\tTcim\n",
      "ENSMUSG00000048489\tDepp1\n",
      "ENSMUSG00000020101\tVsir\n",
      "ENSMUSG00000006191\tCdkal1\n",
      "ENSMUSG00000055555\tCt55\n",
      "ENSMUSG00000025420\tKatnal2\n",
      "ENSMUSG00000041889\tShisa4\n",
      "ENSMUSG00000026163\tSphkap\n",
      "ENSMUSG00000075408\tSmim41\n",
      "ENSMUSG00000045502\tHcar2\n",
      "ENSMUSG00000049241\tHcar1\n",
      "ENSMUSG00000042590\tIpo11\n",
      "ENSMUSG00000051586\tMical3\n",
      "ENSMUSG00000026796\tNiban2\n",
      "ENSMUSG00000052353\tCemip\n",
      "ENSMUSG00000036412\tArsi\n",
      "ENSMUSG00000048534\tJaml\n",
      "ENSMUSG00000047822\tAngptl8\n",
      "ENSMUSG00000108022\tGm7298\n",
      "ENSMUSG00000028195\tCcn1\n",
      "ENSMUSG00000019997\tCcn2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from biomart import BiomartServer\n",
    "server = BiomartServer('http://www.ensembl.org/biomart')\n",
    "dataset = server.datasets['mmusculus_gene_ensembl']\n",
    "\n",
    "attributes = ['ensembl_gene_id','mgi_symbol']\n",
    "filters = {'mgi_id':['MGI:95292',\n",
    "                     'MGI:95537',\n",
    "                     'MGI:106023',\n",
    "                     'MGI:107565',\n",
    "                     'MGI:109382',\n",
    "                     'MGI:88613',\n",
    "                     'MGI:1277133',\n",
    "                     'MGI:1314653',\n",
    "                     'MGI:1858441',\n",
    "                     'MGI:1860303',\n",
    "                     'MGI:1914701',\n",
    "                     'MGI:1915368',\n",
    "                     'MGI:1915982',\n",
    "                     'MGI:1916318',\n",
    "                     'MGI:1918730',\n",
    "                     'MGI:1921298',\n",
    "                     'MGI:1921765',\n",
    "                     'MGI:1922263',\n",
    "                     'MGI:1924234',\n",
    "                     'MGI:1924802',\n",
    "                     'MGI:1924879',\n",
    "                     'MGI:1924967',\n",
    "                     'MGI:1933383',\n",
    "                     'MGI:2441671',\n",
    "                     'MGI:2442377',\n",
    "                     'MGI:2442733',\n",
    "                     'MGI:2442910',\n",
    "                     'MGI:2443629',\n",
    "                     'MGI:2444394',\n",
    "                     'MGI:2670959',\n",
    "                     'MGI:2685484',\n",
    "                     'MGI:3643534',\n",
    "                     'MGI:3645360',\n",
    "                     'MGI:3646928',\n",
    "                     'MGI:3648717',\n",
    "                     ]}\n",
    "\n",
    "response = dataset.search({'attributes':attributes,'filters':filters})\n",
    "print(response.text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8998990f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# WYNIK_Excell\n",
    "df.to_excel(r'C:\\Users\\joasi\\geneBase\\1_pub\\Wyniki\\notResponse_1_pub_2003233.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e171a3da",
   "metadata": {},
   "outputs": [],
   "source": [
    "MGI_EntrezGene.rpt"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
